generate_docstring.py

```diff
@@ -0,0 +1,537 @@
+        link = (f"https://help.3ds.com/{version}/English/DSSIMULIA_Established/SIMACAEKERRefMap/"
+                f"simaker-c-{name}pyc.htm?contextscope=all")
+        return link, f"Check `{class_or_module_name} on help.3ds.com/{version} <{link}>`_."
+
+    @classmethod
+    def _method_or_function_link(
+        cls,
+        version: typing.Literal['2016', '2017', '2018', '2019', '2020', '2020', '2021', '2022'],
+        type: typing.Literal['class', 'module'],
+        class_or_module_name: str,
+        method_or_function_name: str,
+    ) -> tuple[str, str]:
+        """Generate a link to the Abaqus function documentation.
+
+        Parameters
+        ----------
+        version : str
+            The Abaqus version.
+        type : 'class' or 'module'
+            Type of the object, class or module.
+        class_or_module_name : str
+            The name of the module.
+        method_or_function_name : str
+            The name of the function.
+
+        Returns
+        -------
+        str
+            The link to the documentation.
+        str
+            The link with label.
+        """
+        name = f"{type if type == 'gprmodule' else ''}{class_or_module_name.lower()}"
+        if type == 'class' and method_or_function_name == '__init__':
+            method_or_function_name = class_or_module_name
+        link = (f"https://help.3ds.com/{version}/English/DSSIMULIA_Established/SIMACAEKERRefMap/"
+                f"simaker-c-{name}pyc.htm?contextscope=all"
+                f"#simaker-{name}{method_or_function_name.lower()}pyc")
+        if method_or_function_name == class_or_module_name and type == 'class':
+            signature = f"{class_or_module_name}()"
+        else:
+            signature = f"{class_or_module_name}.{method_or_function_name}()"
+        return link, f"Check `{signature} on help.3ds.com/{version} <{link}>`_."
+
+    @classmethod
+    def _add_link_in_class_or_module_docstring(
+        cls,
+        version: typing.Literal['2016', '2017', '2018', '2019', '2020', '2020', '2021', '2022'],
+        type: typing.Literal['class', 'module'],
+        class_or_module_name: str,
+        docstring: str,
+    ) -> str:
+        """Add a link to the Abaqus documentation to the docstring for classes or modules.
+
+        Parameters
+        ----------
+        version : str
+            The Abaqus version.
+        type : 'class' or 'module'
+            Type of the object, class or module.
+        class_or_module_name : str
+            The name of the class or module.
+        docstring : str
+            The docstring.
+
+        Returns
+        -------
+        str
+            The docstring with the link to the Abaqus documentation.
+        """
+        link, link_with_label = cls._class_or_module_link(version, type, class_or_module_name)
+        return docstring.rstrip() + '\n\n' + ' ' * (0 if type == 'module' else 4) + link_with_label
```

Comment:
Maybe another way to present the links in the docstrings.

Reply:
Decorators can be used to modified the docstrings dynamically, but the links can not be detected by static checker, and we have to add the decorator in every module/class/function/method.